egui/src/any/any_map.rs

```diff
@@ -0,0 +1,205 @@
+use crate::any::element::{AnyMapElement, AnyMapTrait};
+use std::any::TypeId;
+use std::collections::HashMap;
+use std::hash::Hash;
+
+/// Stores any object by `Key`.
+#[derive(Clone, Debug)]
+pub struct AnyMap<Key: Hash + Eq>(HashMap<Key, AnyMapElement>);
```

Comment:
Now that this no longer relies on `Id` it would be trivial to move to its own crate - hint, hint, nudge, nudge ;)